### ISSUE TITLE
strategy for implementation specific logging specs 

### ISSUE THREAD

**Issue by colinsurprenant** [elastic/logstash#4253]
The `logstash-core-event` `Event` specs contains tests like this to test for specific logging calls

```ruby
      it "should warn for invalid string format" do
        expect(LogStash::Event::LOGGER).to receive(:warn)
        LogStash::Event.new("@timestamp" => "foo")
      end
```

Obviously these tests don't pass the Java `Event` implementation because, first, logging is not yet implemented in the Java Event but these tests, as they are, will be difficult to support in any alternate implementation because they rely on an implementation detail which is the use of a specific method from a specific inner object constant. 

I am not sure what would be the proper approach to correctly test logging in a non implementation specific way.

An idea would be to use some  form of dependency injection by either passing a logger object to the Event constructor or use a class method to set a logger instance in an Event class variable. That way we could use a logging mock object in specs, or something around these lines. 

I will investigate a bit.

**Comment by colinsurprenant** [comment]
implemented and fixed in #4263

**Comment by colinsurprenant** [comment]
#4263 merged, closing.
